test/functional/fixtures/tty-test.c

```diff
@@ -34,6 +33,18 @@ bool owns_tty(void)
 #endif
 }
 
+static inline bool is_terminal(const FILE * stream1) 
+    __attribute__((always_inline)) __attribute__ ((pure)) __attribute__((warn_unused_result));
+
+/**
+ * Checks if the file pointer is not a terminal like STDIN or STDOUT 
+ * @param[in] stream to be checked
+ * @return bool
+ */
+static inline bool is_terminal(const FILE * stream){
```

Comment:
There are lots of style guide violations and things that won’t work here:

1. You need to include nvim/func_attr.h to have REAL_FATTR available. And correct approach would be using FUNC_ATTR… and no declarations since you are in a C file, but include header with generated declarations. But as files in test/functional/fixtures are compiled differently I am not sure whether “correct” approach will work at all.
2. `fileno()` does not accept pointers to const memory. So argument type should be `FILE *const` or there should be a cast below, but latter is theoretically dangerous.
3. We are not using `/*`-style comments.
4. `{` on the same line as function declaration is not correct either, do you see any other functions below like this?
5. Should have no space after asterisk.
6. Indentation shift witdth is 2, not 4. Applies to two lines here.
7. Do not add useless `@return bool`, doxygen does include function declarations.
8. Two spaces after `[in]`, two spaces after `stream`, blank line between `@param` and summary (also between `@param` and `@return`), and correct sentense in parameter description. Though it is better to not have `@param` as well, only function summary is enough.

And there should not be a big bunch of commits with merges, such small PRs better be rebased.

Reply:
1.  Points 3,4,5,6,7,8 : Will make the relevant changes . However is there any documentation to check into  the same. Will go through the same.
2.  Noted point about the rebasing .. will rebase and recommit the changes.
3. Point 1 : I Used REAL_FATTR_ALWAYS_INLINE in the last commit and got error. so that is the reason for writing the same directly. Or is it ok to directly use FUNC_ATTR_WARN_UNUSED_RESULT

Reply:
\3. For REAL_FATTR_… you need to include nvim/func_attr.h (what I did say right in the first sentence of point 1). FUNC_ATTR_… is most likely not going to work, requires more changes and you will need that include as well as include with generated declarations which I am 90% sure does not exist.

Reply:
And note that “directly use FUNC_ATTR…” is not valid in this context: FUNC_ATTR…, depending on macros which were defined when nvim/func_attr.h was included, may expand to REAL_FATTR…, to nothing or do not expand at all, so you may say that you want to “directly use REAL_FATTR”, “directly use __attribute__(…)” (i.e. what REAL_FATTR… macros expand to) (don’t actually do this), but can’t say the same about FUNC_ATTR… as it is most indirect way to attach attribute to a function possible in this project.

Reply:
I Have made the changes and committed them. Pls review once so that i can take up further on a similar note.